### PR TITLE
802154 phy test use ipc uart

### PR DESCRIPTION
**Pull request by ankuns**
For `nrf5340dk_nrf5340_cpunet` target the shell is directed to
remote_shell. This approach frees serial box of nRFf5340 cpunet and
allows to use other peripherals with the same peripheral id.

This PR contains also bugfixes required this change.

### CODE REVIEW

File: samples/peripheral/802154_phy_test/modules/app_rpc/zephyr/Kconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config APP_RPC
```
Comment (ahasztag) [inline samples/peripheral/802154_phy_test/modules/app_rpc/zephyr/Kconfig:7]: I see that this a general module, which could also be used in other applications, and that is also why this is `APP_RPC` and not `PTT_APP_RPC` - however in this case maybe we should consider moving it, for example to `nrf5340/remote_shell`.

**Review by FrancescoSer — APPROVED** [review]

File: samples/peripheral/802154_phy_test/boards/nrf5340dk_nrf5340_cpunet.overlay [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/ {
```
Comment (jciupis) [inline samples/peripheral/802154_phy_test/boards/nrf5340dk_nrf5340_cpunet.overlay:1]: Missing license

File: samples/peripheral/802154_phy_test/boards/nrf21540dk_nrf52840.conf [hunk 3]
```diff
@@ -0,0 +1 @@
+CONFIG_SHELL_PROMPT_UART=">"
```
Comment (jciupis) [inline samples/peripheral/802154_phy_test/boards/nrf21540dk_nrf52840.conf:1]: Missing license. I know this file contains one trivial line but I'm assuming we always have to publish licensed files.

**Review by jciupis — APPROVED** [review]
Looks good! Two minor comments below.

**Review by ahasztag — APPROVED** [review]
I've left just one comment just to consider, however apart from that everythings seems ok

File: samples/peripheral/802154_phy_test/boards/nrf5340dk_nrf5340_cpunet.overlay (hunk 2 above)
Comment (ankuns) [reply samples/peripheral/802154_phy_test/boards/nrf5340dk_nrf5340_cpunet.overlay:7]: amended

File: samples/peripheral/802154_phy_test/boards/nrf21540dk_nrf52840.conf (hunk 3 above)
Comment (ankuns) [reply samples/peripheral/802154_phy_test/boards/nrf21540dk_nrf52840.conf:7]: amended

**Pull request merged**